core/src/main/java/org/infinispan/commands/RemoveCacheCommand.java

```diff
@@ -23,31 +18,19 @@
    public static final byte COMMAND_ID = 18;
 
    private EmbeddedCacheManager cacheManager;
-   private GlobalComponentRegistry registry;
-   private PersistenceManager persistenceManager;
-   private CacheJmxRegistration cacheJmxRegistration;
 
    private RemoveCacheCommand() {
       super(null); // For command id uniqueness test
    }
 
-   public RemoveCacheCommand(String cacheName, EmbeddedCacheManager cacheManager,
-         GlobalComponentRegistry registry, PersistenceManager persistenceManager,
-         CacheJmxRegistration cacheJmxRegistration) {
+   public RemoveCacheCommand(String cacheName, EmbeddedCacheManager cacheManager) {
       super(cacheName);
       this.cacheManager = cacheManager;
-      this.registry = registry;
-      this.persistenceManager = persistenceManager;
-      this.cacheJmxRegistration = cacheJmxRegistration;
    }
 
    @Override
    public Object perform(InvocationContext ctx) throws Throwable {
-      persistenceManager.setClearOnStop(true);
-      cacheJmxRegistration.setUnregisterCacheMBean(true);
-      Cache<Object, Object> cache = cacheManager.getCache(cacheName);
-      cache.stop();
-      registry.removeCache(cacheName);
+      ((DefaultCacheManager) cacheManager).removeCacheLocally(cacheName);
```

Comment:
> I need to access the private field in DefaultCacheManager (configurationOverrides and cacheDependencyGraph)

Ok. I'd suggest you do the following:

1. Check with Paul Ferraro if your changes would be problematic for him. I suspect that if you did the cast to `DefaultCacheManager`, then they would if they used this operation. If it's not a problem, then go ahead with the current fix.

2. If Paul says it could be problematic, I still don't think we should expose this `removeCacheLocally` via EmbeddedCacheManager, so what I'd try to do is to break the dependency between DefaultCacheManager and the `configurationOverrides` and `cacheDependencyGraph`. These two could be moved to a global component that deals with those.

How does that sound?